mkosi/__init__.py

```diff
@@ -4558,21 +4562,17 @@ def run_verb(args: Args, images: Sequence[Config], *, resources: Path) -> None:
     build = False
 
     for i, config in enumerate(images):
-        tools = (
-            finalize_default_tools(args, config, resources=resources)
-            if config.tools_tree and config.tools_tree == Path("default")
-            else None
-        )
+        if tools and config.tools_tree == Path("default"):
+            tools_tree = tools.output_dir_or_cwd() / tools.output
+        else:
+            tools_tree = config.tools_tree
```

Comment:
mypy can't synthesize the type properly because the two branches have different types
```suggestion
        tools_tree = (
            tools.output_dir_or_cwd() / tools.output
            if tools and config.tools_tree == Path("default")
            else config.tools_tree
        )
```